### DISCUSSION TITLE
Is it possible to make this work without npm but with importmap

### DISCUSSION THREAD

**Discussion by troptropcontent** [primer/view_components#2135] [Q&A] [answered]
Hello :) 

I am pretty new to javascript, i have a rails 7 that does not have npm or yarn installed and that is for now only using importmaps. 

I would like to use the primer CSS to support the view components. 

Is it possible to have primer/css without npm, only using importmaps ? 

Thank you so much for your help, 
Tom

**Comment by camertron** [comment, marked as answer]
Hey @troptropcontent, interesting question! I have personally never used import maps before, but after reading a bit about them I think primer/view-components should work with them. Try putting something like this in config/importmap.rb:

```ruby
pin "@primer/view-components", to: "https://ga.jspm.io/npm:@primer/view-components@0.3.1/app/assets/javascripts/primer_view_components.js"
```

As for the CSS from primer/view-components and primer/css, you should be able to pull them in directly using `<link>` tags:

```html
<link rel="stylesheet" href="https://unpkg.com/@primer/view-components@0.3.1/app/assets/styles/primer_view_components.css" crossorigin="" />
<link rel="stylesheet" href="https://unpkg.com/@primer/css@21.0.7/dist/primer.css" crossorigin="" />
```

**Reply by troptropcontent** [reply]
It worked perfectly 🙏 

Thank you so much !!

One more question, i have a date input to setup, but i do not see a date_input in the form documentation 🙏 

How would you do that ?

**Reply by camertron** [reply]
Awesome, glad it's working!

> One more question, i have a date input to setup, but i do not see a date_input in the form documentation

Ah yeah, date inputs aren't currently supported by the forms framework. A member of our team was actually looking at adding one last week, not sure how much progress was made. I wrote up [this gist](https://gist.github.com/camertron/22c0880171226c82392271318f48cda9) as a guide for adding new input types, although it's not much to go on 😅 I have a message out to my my team member and will post here when I hear back. You're welcome to take a shot at it too if you like 👍

**Reply by troptropcontent** [reply]
I will definitely give it a try 🙏 

(I think that the gist link you provide is not the good one, it redirects to here)

**Reply by camertron** [reply]
Woops, sorry about that! Just edited the post to include the right link 😅

**Comment by dereke** [comment]
I notice this in the guides - https://primer.style/product/getting-started/rails/#import-maps

> Unfortunately at the time of this writing (June 2024), it is not possible to use primer_view_components with import maps. A number of primer_view_component's JavaScript dependencies attempt to import paths that the import map isn't expecting (eg. tab-container-element).

What is the state as of Aug 2025?

**Comment by camertron** [comment]
Hey there @dereke 👋 

Unfortunately I am no longer with GitHub, but given my past experience working on this library and my knowledge of the dynamics of the Primer team, I am 99% sure nothing has changed with regard to using PVC with importmaps. That is to say, it is still not possible to use PVC with importmaps for the reason mentioned in the documentation.

One way to avoid the import issue might be to include individual compiled javascript files, one per component, to the build output. You could then pull in the components you actually need instead of the whole thing, and hopefully you only need components that don't attempt to import 3rd party modules. That's really just a band-aid though.
